sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Primitives/ServiceBusSessionMessageActions.cs

```diff
@@ -66,5 +68,11 @@ protected ServiceBusSessionMessageActions()
                 await _eventArgs.SetSessionStateAsync(sessionState, cancellationToken).ConfigureAwait(false);
             }
         }
+
+        /// <inheritdoc cref="ProcessSessionMessageEventArgs.ReleaseSession()"/>
+        public virtual void ReleaseSession()
+        {
+            IsReleaseSession = true;
```

Comment:
Out of curiosity, why do we wait to call release session on the args until after return to the listener?   Is that to avoid an error if the session is released before attempting to settle the message?

Reply:
Yeah, technically I could call args.ReleaseSession here, but for the receiver case, we would need to close the receiver after the function returns, so I just decided to handle both cases with the same property.